### PR TITLE
Like button and View Recipe functionality

### PR DESCRIPTION
**Pull request by NicholasKao1029**
Initial PR for like button functionality

### CODE REVIEW

File: tmp.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+POST /user_activity/
```
Comment (Harin329) [inline tmp.txt:1]: was this file just for your own reference? maybe we can remove this from the commit

File: frontend/src/screens/tab1/recipe.js [hunk 2]
```diff
@@ -135,23 +143,27 @@ export default function Recipe({navigation, route}) {
...
-              style={{
-                width: 24,
-                height: 24,
-                resizeMode: 'contain',
-                marginRight: 5,
-              }}
-            />
+            }}
+          >
+            <TouchableOpacity activeOpacity={0.5} onPress={this.Func}>
```
Comment (Harin329) [inline frontend/src/screens/tab1/recipe.js:156]: So this on press is where your logic should trigger the endpoint. A dispatch should be called with your action (POST_RECIPE_LIKE) and the param. So something along the lines of onPress={() => {dispatch({type: GET_RECIPE, recipe_id: recipe.recipe_id})}}, but naming the `recipe_id` param part accordingly to the saga function (recipe_like_id).

Comment (Harin329) [reply frontend/src/screens/tab1/recipe.js:156]: a suggestion would be to just make a mapping like { recipe_like_id: 'id', user_id: 'id'} and attach that object as a part of the dispatch so u can access multiple params

**Review by Harin329 — CHANGES_REQUESTED** [review]

File: frontend/src/screens/tab1/recipe.js (hunk 2 above)
Comment (NicholasKao1029) [reply frontend/src/screens/tab1/recipe.js]: Currently doing

```
                dispatch({
                  type: POST_RECIPE_LIKE,
                  user_id: auth().currentUser.uid,
                  recipe_like_id: recipe.recipe_id,
                });
```

Does it not work with more keys than one?

**Review by Harin329 — APPROVED** [review]

**Pull request merged**